sdk/python/requirements.in

```diff
@@ -34,3 +34,4 @@ fire>=0.3.1,<1
 google-api-python-client>=1.7.8,<2
 dataclasses>=0.8,<1; python_version<"3.7"
 pydantic>=1.8.2,<2
+typing-extensions>=3.10.0.2,<4
```

Comment:
Should add this in setup.py as well, we can't rely on an indirect dependency from pydantic.
Also this better be conditioned by a Python version, 3.9 I think?
See dataclass for example:
https://github.com/kubeflow/pipelines/blob/55a2fb5c20011b01945c9867ddff0d39e9db1964/sdk/python/setup.py#L53-L54